src/lxml/tests/test_external_document.py

```diff
@@ -3,17 +3,93 @@
 Test cases related to direct loading of external libxml2 documents
 """
 
-import unittest
 import sys
+import unittest
 
-from common_imports import etree, HelperTestCase
+from common_imports import HelperTestCase, etree
+
+DOC_NAME = b'libxml2:xmlDoc'
+DESTRUCTOR_NAME = b'destructor:xmlFreeDoc'
 
 
 @unittest.skipIf(sys.version_info[:2] < (2, 7),
                  'Not supported for python < 2.7')
 class ExternalDocumentTestCase(HelperTestCase):
-    def test_external_document_type_checking(self):
+    def setUp(self):
+        import ctypes
+        from ctypes import pythonapi
+        from ctypes.util import find_library
+
+        def wrap(func, restype, *argtypes):
```

Comment:
To expand on why it is natural -- when you read a function declaration in C the result type is before the argument types.